### PR TITLE
Issue 472 variable viewer with values of type string

### LINKED ISSUE #472: Allow String variables to be displayed in Variable Viewer
**Is your feature request related to a problem? Please describe.**
The variable viewer does not display a variable if its value is a string.

For example, the following variable in the xml file will not appear in the variable viewer:
`
<propeller>
          <manufacturer is_input="False">APC</manufacturer >
`

**Describe the solution you'd like**
In the `VariableList` class, the  `to_dataframe` method could be modified to allow for non-float values (line `value = self._as_list_or_float(variable.value)`) :

```
def to_dataframe(self) -> pd.DataFrame:
        
        var_dict = {"name": []}
        var_dict.update({metadata_name: [] for metadata_name in self.metadata_keys()})

        for variable in self:
            value = self._as_list_or_float(variable.value)  # Allow for non-float values
            var_dict["name"].append(variable.name)
            for metadata_name in self.metadata_keys():
                if metadata_name == "val":
                    var_dict["val"].append(value)
                else:
                    # TODO: make this more generic
                    if metadata_name in ["val", "initial_value", "lower", "upper"]:
                        metadata = self._as_list_or_float(variable.metadata[metadata_name])
                    else:
                        metadata = variable.metadata[metadata_name]
                    var_dict[metadata_name].append(metadata)

        df = pd.DataFrame.from_dict(var_dict)

        return df
```

### PR DESCRIPTION
**Pull request by ScottDelbecq**
This PR closes #472 by adding the possibility to have variables with value of type string. This is particularly useful when using discrete variables to store information for displaying of computation logics.

### CODE REVIEW

**Review by christophe-david — COMMENTED** [review]
Thanks for this work (and for having paid attention to the order of commits 😉).

A few changes needed, especially because the modifications of `variable_list.py` have apparently no use.

For the record, we have a problem, prior to this PR, that prevents to run `test_variables.py` separately from the other tests.
A quick fix will be to instantiate `FastoadLoader()` at the beginning to the test. You may do that if it is convenient for you, but I'll address this problem in a better manner right after the merge of this PR.

File: src/fastoad/openmdao/variables/variable_list.py [hunk 1]
```diff
@@ -153,15 +153,17 @@ def to_dataframe(self) -> pd.DataFrame:
         var_dict.update({metadata_name: [] for metadata_name in self.metadata_keys()})
 
         for variable in self:
-            value = self._as_list_or_float(variable.value)
+            value = self._as_list_or_float_or_str(variable.value)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/openmdao/variables/variable_list.py]: Corrected in fd8de3e and 5ae2e73.

File: src/fastoad/openmdao/variables/variable.py [hunk 2]
```diff
@@ -328,7 +328,11 @@ def __eq__(self, other):
         return (
             isinstance(other, Variable)
             and self.name == other.name
-            and np.all(np.isclose(my_value, other_value, equal_nan=True))
+            # First condition for val of type string
+            and (
+                np.all(my_value == other_value)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/openmdao/variables/variable.py]: Corrected in 8b8c6e0.

File: src/fastoad/openmdao/tests/test_variables.py [hunk 3]
```diff
@@ -154,14 +154,17 @@ def test_df_from_to_variables():
...
-    assert np.all(df["units"].to_list() == [None, "m", "kg/s"])
-    assert np.all(df["desc"].to_list() == ["", "", "some test"])
+    assert np.all(df["name"] == ["a", "b", "c", "d"])
+    assert np.all(df["val"] == [5, [1.0, 2.0, 3.0], [1.0, 2.0, 3.0], "my value is a string"])
+    assert np.all(df["units"].to_list() == [None, "m", "kg/s", None])
+    assert np.all(df["desc"].to_list() == ["", "", "some test", ""])
 
     new_vars = VariableList.from_dataframe(df)
+    print(vars)
+    print(new_vars)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/openmdao/tests/test_variables.py]: Corrected in f5b67ef

**Review by christophe-david — APPROVED** [review]
All done and dusted.
Thank you.

**Pull request merged**